lib/pages/chat/chat_view.dart

```diff
@@ -115,15 +116,8 @@ class ChatView extends StatelessWidget {
     controller.sendingClient ??= client;
     controller.room = controller.sendingClient!.getRoomById(controller.roomId!);
     if (controller.room == null) {
-      return Scaffold(
-        appBar: AppBar(
-          toolbarHeight: 120,
-          title: Text(L10n.of(context)!.oopsSomethingWentWrong),
-        ),
-        body: Center(
-          child: Text(L10n.of(context)!.youAreNoLongerParticipatingInThisChat),
-        ),
-      );
+      context.go('/error');
+      return const SizedBox.shrink();
     }
```

Comment:
IMO: Move it to `initState`.

Reply:
<img width="410" alt="Screenshot 2023-09-19 at 08 28 09" src="https://github.com/linagora/twake-on-matrix/assets/99852347/4782c68f-121e-414f-89e9-37ef018203a2">

Reply:
why is it better for you ?

Reply:
Using Navigator inside the build method of a Flutter widget is generally not recommended and can lead to unexpected behavior or errors. The build method should be responsible for describing the UI hierarchy based on the current state and should not trigger navigation actions directly. Instead, you should initiate navigation in response to user interactions or specific events.

Here's why it's not a good practice:

Infinite Loop: Placing navigation logic inside the build method can lead to an infinite loop. When Navigator is called, it triggers a rebuild of the widget tree. If the navigation action itself causes the widget to rebuild (for example, because it changes the route or state), it can lead to an infinite loop of rebuilds.

Build Method Frequency: The build method can be called frequently, even when there are no user interactions. Placing navigation logic inside build might cause unintended navigation actions to occur frequently.

Reply:
@nqhhdev  agree with you

Reply:
done